mds/admin.py

```diff
@@ -10,15 +20,38 @@ class ProviderAdmin(admin.ModelAdmin):
 
 @admin.register(models.Device)
 class DeviceAdmin(admin.ModelAdmin):
-    list_display = ["id", "provider", "identification_number", "model"]
-    list_filter = ["provider", "model"]
+    list_display = ["id", "provider", "identification_number", "category"]
+    list_filter = ["provider", "model", "category"]
+    search_fields = ["id", "identification_number"]
 
 
 @admin.register(models.EventRecord)
 class EventRecordAdmin(admin.ModelAdmin):
-    list_display = ["timestamp", "device", "event_type"]
+    list_display = ["timestamp", "provider", "device_id", "event_type"]
     list_filter = ["device__provider", "event_type"]
-    ordering = ["-timestamp"]
+    list_select_related = ("device__provider",)
+    search_fields = ["device__id", "device__identification_number"]
+
+    def get_search_results(self, request, queryset, search_term):
+        if not search_term:
+            return super().get_search_results(request, queryset, search_term)
+        custom_queryset = get_devices_queryset_search_results(self, search_term)
+        return super().get_search_results(request, custom_queryset, search_term)
```

Comment:
```suggestion
        queryset, use_distinct = super().get_search_results(request, queryset, search_term)
        queryset |= self.model.objects.filter(device__identification_number=search_term)
        if is_uuid(search_term):
             queryset |= self.model.objects.filter(device_id=search_term)
        return queryset, use_distinct
```

Reply:
Hmm I need to use the custom function get_devices_queryset_search_results  in other admin models too, so I have to create a separate function

Reply:
uh ? I don't get how other admin models would use it since the `select_related`, filters  on `device_id` and `device__identification_number` are specific to mds.models.EventRecord

Reply:
No, you can use it on Keys too

Reply:
OK I see. It's not a part of django-mds though. And we'll probably want to customize search for keys (like area name).
My personal criteria on when to factorize: when use case is exactly the same, a bit tricky in at least 3 different places in the code.

Reply:
In fact, i will use it on keys too. It will be used in 2 different places. As it will be used outside of django-mds too, I prefer to factorize here.